gitbook/audio-recording.md

```diff
@@ -2,125 +2,119 @@
+Audio samples are provided to the `onProcessSamples` function every 100 ms. This is currently hard-coded into CrSFML and you can't change that (unless you modify CrSFML itself). This may change in the future.
 
-There are also two additional virtual functions that you can optionally override: `onStart` and `onStop`. They are called when the capture starts/stops respectively. They are useful for initialization/cleanup tasks. 
+There are also two additional virtual functions that you can optionally override: `onStart` and `onStop`. They are called when the capture starts/stops respectively. They are useful for initialization/cleanup tasks.
 
-Here is the skeleton of a complete derived class: 
+Here is the skeleton of a complete derived class:
 
-```
-class MyRecorder : public sf::SoundRecorder
-{
-    virtual bool onStart() // optional
-    {
-        // initialize whatever has to be done before the capture starts
+```ruby
+class MyRecorder < SF::SoundRecorder
+    def on_start # optional
+        # initialize whatever has to be done before the capture starts
         ...
 
-        // return true to start the capture, or false to cancel it
-        return true;
-    }
+        # return true to start the capture, or false to cancel it
+        true
+    end
 
-    virtual bool onProcessSamples(const Int16* samples, std::size_t sampleCount)
-    {
-        // do something useful with the new chunk of samples
+    def on_process_samples(samples, sample_count)
+        # do something useful with the new chunk of samples
         ...
 
-        // return true to continue the capture, or false to stop it
-        return true;
-    }
+        # return true to continue the capture, or false to stop it
+        true
+    end
 
-    virtual void onStop() // optional
-    {
-        // clean up whatever has to be done after the capture is finished
+    def on_stop # optional
+        # clean up whatever has to be done after the capture is finished
         ...
-    }
-}
+    end
+end
 ```
 
-The `isAvailable`/`start`/`stop` functions are defined in the [SoundRecorder]({{book.api}}/SoundRecorder.html) base, and thus inherited in every derived classes. This means that you can use any recorder class exactly the same way as the [SoundBufferRecorder]({{book.api}}/SoundBufferRecorder.html) class above. 
+The `is_available`/`start`/`stop` functions are defined in the [SoundRecorder]({{book.api}}/SoundRecorder.html) base, and thus inherited in every derived classes. This means that you can use any recorder class exactly the same way as the [SoundBufferRecorder]({{book.api}}/SoundBufferRecorder.html) class above.
 
-```
-if (!MyRecorder::isAvailable())
-{
-    // error...
-}
+```ruby
+unless MyRecorder.is_available()
+    # error...
+end
 
-MyRecorder recorder;
-recorder.start();
+recorder = MyRecorder.new
```

Comment:
Ok. If someday someone find a good reason to use it, then he probably will open an issue and _then_ it can be fixed :stuck_out_tongue: